biosynseq/generate.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import os
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict, List
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from gene_transformer.config import ModelSettings
+from gene_transformer.model import (
+    LoadDeepSpeedStrategy,
+    LoadPTCheckpointStrategy,
+    ModelLoadStrategy,
+    inference,
+)
+from gene_transformer.utils import non_redundant_generation, seqs_to_fasta
+
+logger = logging.getLogger(__name__)
+
+
+def generate_fasta(
+    model_strategy: ModelLoadStrategy, fasta_path: str, num_seqs: int
+) -> Dict[str, List[str]]:
+    """Given pt file or deepspeed directory, output generated sequences' fasta file.
+
+    Parameters
+    ----------
+    model_strategy : ModelLoadStrategy
+        Model used to generate sequences, depending on whether a pt file or a deepspeed directory is given.
+    fasta_path : str
+        Path to save fasta sequences.
+    num_seqs : int
+        Number of non-redundant sequences to generate.
+
+    Returns
+    -------
+    Dict[str, List[str]]
+        Unique generated sequences.
+    """
+
+    # obtain model
+    model = model_strategy.get_model()
+    model.cuda()
+    # generate non-redundant sequences
+    results = non_redundant_generation(
+        model=model.model, tokenizer=model.tokenizer, num_seqs=num_seqs
+    )
+    results["unique_seqs"] = list(results["unique_seqs"])
+    # turn unique sequences to fasta
+    unique_seqs = results.get("unique_seqs")
+    seqs_to_fasta(seqs=unique_seqs, file_name=fasta_path)
+    return results
+
+
+def fasta_to_embeddings(
+    model_strategy: ModelLoadStrategy, fasta_path: str, embeddings_output_path: str
+) -> np.ndarray:
+    """Run inference to generate embeddings for generated sequences.
+
+    Parameters
+    ----------
+    model_strategy : ModelLoadStrategy
+        Model used to generate sequences, depending on whether a pt file or deepspeed is given.
+    fasta_path : str
+        Path to access fasta sequences.
+    embeddings_output_path : str
+        Path to save generated embeddings.
+
+    Returns
+    -------
+    np.ndarray
+        Generated embeddings.
+    """
+
+    # run inference
+    embeddings = inference(model_strategy, fasta_path, embeddings_output_path)
+    return embeddings
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="Generate sequences or embeddings.")
+    parser.add_argument(
+        "-c",
+        "--config",
+        type=str,
+        required=True,
+        help=".yaml file containing configuration settings",
+    )
+    parser.add_argument(
+        "--mode",
+        default="get_fasta",
+        type=str,
+        required=True,
+        help="get_fasta or get_embeddings",
+    )
+    parser.add_argument(
+        "--pt_path", type=str, required=True, help="Path to access pt file"
+    )
+    parser.add_argument(
+        "--fasta_path",
+        default="",
+        type=str,
```

Comment:
We can change type to Path

Reply:
Make sure to update the docstring and type hints in the functions that use it as well!